### PR TITLE
[query] Add tests for local backend

### PR DESCRIPTION
**Pull request by tpoterba**

### CODE REVIEW

File: build.yaml [hunk 1]
```diff
@@ -1139,6 +1139,216 @@ steps:
      - batch_pods_ns
      - hail_run_image
      - build_hail
+ - kind: runImage
```
Comment (tpoterba) [inline build.yaml:1142]: changes to build.yaml involve copying the python test tasks and adding 
```
export HAIL_QUERY_BACKEND=local
```

File: hail/python/test/hail/helpers.py [hunk 2]
```diff
@@ -127,6 +128,19 @@ def wrapper(func, *args, **kwargs):
...
+
+    @decorator
+    def wrapper(func, *args, **kwargs):
+        if isinstance(hl.utils.java.Env.backend(), LocalBackend):
+            with pytest.raises(BaseException):
+                return func(*args, **kwargs)
+        else:
+            return func(*args, **kwargs)
+
+    return wrapper
```
Comment (patrick-schultz) [inline hail/python/test/hail/helpers.py:142]: It looks like there's an official pytest decorator to mark a test as expected to fail, which the test reports treat specially. https://docs.pytest.org/en/latest/skipping.html#xfail-mark-test-functions-as-expected-to-fail

Using that, I think you could write this
```
fails_local_backend = pytest.mark.fail(isinstance(hl.utils.java.Env.backend(), LocalBackend), reason="doesn't yet work on local backend", strict=True)
```

File: build.yaml [hunk 3]
```diff
@@ -1139,6 +1139,216 @@ steps:
...
+     tar xzf test.tar.gz
+     tar xzf resources.tar.gz
+     tar xzf data.tar.gz
+     tar xvf wheel-container.tar
+     python3 -m pip install --no-dependencies hail-*-py3-none-any.whl
+     export HAIL_TEST_RESOURCES_DIR=./resources
+     export HAIL_DOCTEST_DATA_DIR=./data
+     export HAIL_TEST_BUCKET=hail-test-dmk9z
+     export GOOGLE_APPLICATION_CREDENTIALS=/test-gsa-key/key.json
+     export PYSPARK_SUBMIT_ARGS="--driver-memory 6g pyspark-shell"
```
Comment (patrick-schultz) [inline build.yaml:1161]: I'm guessing some of these aren't needed in local mode. Is it worth deleting them to keep things tidy?

**Review by patrick-schultz — CHANGES_REQUESTED** [review]
Nice! Just a couple comments.

File: build.yaml (hunk 3 above)
Comment (tpoterba) [reply build.yaml:1161]: oh, I can delete the pyspark one

File: hail/python/test/hail/helpers.py (hunk 2 above)
Comment (patrick-schultz) [reply hail/python/test/hail/helpers.py:142]: Thoughts on this? It seems nice to have the pytest report list separate counts for “expected to fail” and “unexpectedly passing”.

Comment (tpoterba) [reply hail/python/test/hail/helpers.py:142]: oh, nice! I couldn't find this when I was poking around for something better.

Comment (tpoterba) [reply hail/python/test/hail/helpers.py]: using xfail now, way better!

File: build.yaml [hunk 4]
```diff
@@ -1242,7 +1236,6 @@ steps:
      export HAIL_DOCTEST_DATA_DIR=./data
      export HAIL_TEST_BUCKET=hail-test-dmk9z
      export GOOGLE_APPLICATION_CREDENTIALS=/test-gsa-key/key.json
-     export PYSPARK_SUBMIT_ARGS="--driver-memory 6g pyspark-shell"
```
Comment (patrick-schultz) [inline build.yaml]: I'm assuming it's a mistake that only `test_hail_python_local_backend_2` still has HAIL_TEST_BUCKET and GOOGLE_APPLICATION_CREDENTIALS?

**Review by patrick-schultz — DISMISSED** [review]

**Review by patrick-schultz — APPROVED** [review]

**Pull request merged**